### PR TITLE
[flutter_tools] update libimobiledevice

### LINKED ISSUE #59506: Update packaged libimobiledevice to latest
- [libimobiledevice 1.3.0](https://github.com/libimobiledevice/libimobiledevice/releases/tag/1.3.0)
- [libusbmuxd 2.0.2](https://github.com/libimobiledevice/libusbmuxd/releases/tag/2.0.2)
- [libplist 2.2.0](https://github.com/libimobiledevice/libplist/releases/tag/2.2.0)

Test newest versions with the tool. If all looks good, package it up and ship it.

Also fix https://github.com/flutter/flutter/issues/50299

### PR DESCRIPTION
**Pull request by christopherfujino**
## Description

|package|release|hash|build|codesigned|
|-|-|-|-|-|
|libimobiledevice|1.3.0|569016da35ccea9d9b7c81954342fabb98149fb5|[build](https://ci.chromium.org/p/flutter/builders/prod/libimobiledevice/16)|yes|
|libusbmuxd|2.0.2|ce98c346b7c1dc2a21faea4fd3f32c88e27ca2af|[build](https://ci.chromium.org/p/flutter/builders/prod/usbmuxd/4)|yes|
|libplist|2.2.0|20a2f8dbddcf1a96ad4c720b9afd1d0876d17ffc|[build](https://ci.chromium.org/p/flutter/builders/prod/libplist/13)|yes|
|libtasn1|4.16.0|78204593822b4c1a8e0ca87f57c48e5936b0fe2c|n/a|n/a|
|openssl|1.1.1g|e2e09d9fba1187f8d6aafaa34d4172f56f1ffb72|[build](https://ci.chromium.org/p/flutter/builders/prod/openssl/4)|yes|
|libusb|1.0.23|e782eeb2514266f6738e242cdcb18e3ae1ed06fa|n/a|n/a|

## Related Issues

Fixes https://github.com/flutter/flutter/issues/59506

## Tests

I added the following tests:

*Replace this with a list of the tests that you added as part of this PR. A change in behaviour with no test covering it
will likely get reverted accidentally sooner or later. PRs must include tests for all changed/updated/fixed behaviors. See [Test Coverage].*

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I signed the [CLA].
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] All existing and new tests are passing.
- [ ] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [ ] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by jmagman** [comment]
I locally smoke tested upgraded libimobiledevice, libusbmuxd, libplist (not openssl, etc) and it seemed to all work.

File: packages/flutter_tools/lib/src/ios/devices.dart [hunk 1]
```diff
@@ -823,8 +823,8 @@ class IOSDevicePortForwarder extends DevicePortForwarder {
       process = await _processUtils.start(
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/ios/devices.dart:823]: can you either update (or delete) the usage comment?

**Comment by christopherfujino** [comment]
I should land https://github.com/flutter/flutter/pull/60391 before this one.

**Review by christopherfujino — COMMENTED** [review]
@jmagman your changes LGTM. Can I get your approval too?

**Review by jmagman — APPROVED** [review]
LGTM 🤞

**Comment by christopherfujino** [comment]
Reverted because of https://github.com/flutter/flutter/issues/60559

**Pull request merged**